src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -15,30 +22,66 @@
   /** Store class that gives access to Users. */
   private UserStore userStore;
 
+  /** Store class that gives access to Conversations. */
+  private ConversationStore conversationStore;
+
   /** Set up state for handling activty feed requests. */
   @Override
   public void init() throws ServletException {
   	super.init();
     setUserStore(UserStore.getInstance());
+    setConversationStore(ConversationStore.getInstance());
   }
 
   /**
-   * Sets the UserStore used by this servlet. This function provides a common setup method for use
-   * by the test framework or the servlet's init() function.
+   * Sets the UserStore used by this servlet. This function provides a common 
+   * setup method for use by the test framework or the servlet's init() function.
    */
   void setUserStore(UserStore userStore) {
     this.userStore = userStore;
   }
 
+  /**
+   * Sets the ConversationStore used by this servlet. This function provides a 
+   * common setup method for use by the test framework or the servlet's init() function.
+   */
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+
   /**
    * This function fires when a user navigates to the activity feed page. It 
    * simply forwards to activityfeed.jsp for rendering.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) 
       throws IOException, ServletException {
+    // Get and set Users
     List<User> users = userStore.getAllUsers();
     request.setAttribute("users", users);
+
+    // Get and set Conversations
+    List<Conversation> conversations = conversationStore.getAllConversations();
+    request.setAttribute("conversations", conversations);
+
+    // Make, sort, and set the event map
```

Comment:
If we care about having conversations in order, I wonder if we want to change the way they're stored. Think about having *a lot* of conversations in the database, are we going to sort them all on every request?

This is an exploratory question though, it's fine if we start with a simple implementation for now and then iterate to make it better.

Reply:
I was thinking of sorting them as they're created, but then when I combined the conversations with users I didn't go through with that. I'll look into it with the Event class you mentioned below.